voluptuous_openapi/__init__.py

```diff
@@ -108,15 +109,41 @@ def ensure_default(value: dict[str:Any]):
             pval = ensure_default(pval)
 
             if isinstance(pkey, vol.Any):
-                for val in pkey.validators:
-                    if isinstance(val, vol.Marker):
-                        if val.description:
-                            properties[str(val.schema)] = pval.copy()
-                            properties[str(val.schema)]["description"] = val.description
+                # Handle Required(Any(...)) pattern for anyOf constraints
+                if isinstance(key, vol.Required):
+                    # Extract candidate keys from Any validator
+                    candidate_keys = []
+                    for val_item in pkey.validators:
+                        if isinstance(val_item, vol.Marker):
+                            candidate_keys.append(str(val_item.schema))
                         else:
-                            properties[str(val)] = pval
+                            candidate_keys.append(str(val_item))
+                    
+                    # Check if the value is object (wildcard for presence-only validation)
+                    if value is object:
+                        # For presence-only validation, don't add properties with types
+                        # Just add the constraint group for anyOf generation
+                        any_of_constraint_groups.append(candidate_keys)
                     else:
-                        properties[str(val)] = pval
+                        # Add each candidate key as a property with the specified type
+                        for candidate_key in candidate_keys:
+                            properties[candidate_key] = pval.copy()
+                            if description:
+                                properties[candidate_key]["description"] = description
+                        
```

Comment:
Within the `if isinstance(key, vol.Required):` scope: This feels like it should be inlined in the previous for loop. i'm not sure that these two blocks are needed or if the single loop can be used to figure this whole thing out.